src/domain/youthProfile/hooks/useCreateProfiles.ts

```diff
@@ -0,0 +1,154 @@
+    ADD_SERVICE_CONNECTION,
+    {
+      refetchQueries: ['HasYouthProfile'],
+      awaitRefetchQueries: true,
+    }
+  );
+
+  const [createMyProfile, { loading: creatingMyProfile }] = useMutation<
+    CreateMyProfileData,
+    CreateMyProfileVariables
+  >(CREATE_MY_PROFILE);
+
+  const [
+    createMyYouthProfile,
+    { loading: creatingMyYouthProfile },
+  ] = useMutation<CreateMyYouthProfileData, CreateMyYouthProfileVariables>(
+    CREATE_MY_YOUTH_PROFILE
+  );
+
+  const [updateMyYouthProfile, { loading: updatingMyProfile }] = useMutation<
+    UpdateMyProfileData,
+    UpdateMyProfileVariables
+  >(UPDATE_MY_PROFILE);
+
+  const isLoading =
+    addingServiceConnection ||
+    creatingMyProfile ||
+    creatingMyYouthProfile ||
+    updatingMyProfile;
+
+  const connectService = () => {
+    const serviceConnectionVariables: AddServiceConnectionVariables = {
+      input: {
+        serviceConnection: {
+          service: {
+            type: ServiceType.YOUTH_MEMBERSHIP,
+          },
+          enabled: true,
+        },
+      },
+    };
+
+    addServiceConnection({ variables: serviceConnectionVariables })
+      .then(() => {
+        history.push('/');
+      })
+      .catch((error: Error) => {
+        Sentry.captureException(error);
+        setError(error);
+      });
+  };
+
+  const createYouthProfile = (formValues: FormValues) => {
+    const myYouthProfileVariables: CreateMyYouthProfileVariables = {
+      input: {
+        youthProfile: getCreateYouthProfile(formValues),
+        profileApiToken: profileApiToken,
+      },
+    };
+
+    createMyYouthProfile({ variables: myYouthProfileVariables })
+      .then(result => {
+        if (result.data) {
+          trackEvent({
+            category: 'action',
+            action: 'Register youth membership',
+          });
+          connectService();
+        }
+      })
+      .catch((error: Error) => {
+        Sentry.captureException(error);
+        setError(error);
+      });
+  };
+
+  const createProfiles = (
+    formValues: FormValues,
+    prefillRegistration: PrefillRegistartion
+  ) => {
+    const myProfileVariables: CreateMyProfileVariables = getMutationVariables(
+      formValues,
+      prefillRegistration
+    );
+
+    if (prefillRegistration?.myProfile) {
+      updateMyYouthProfile({ variables: myProfileVariables })
+        .then(result => {
+          if (result.data) createYouthProfile(formValues);
+        })
+        .catch((error: Error) => {
+          Sentry.captureException(error);
+          setError(error);
+        });
+    } else {
+      createMyProfile({ variables: myProfileVariables })
+        .then(result => {
+          if (result.data) createYouthProfile(formValues);
+        })
+        .catch((error: Error) => {
+          Sentry.captureException(error);
+          setError(error);
+        });
+    }
+  };
+
+  return { createProfiles, isLoading, error, setError };
```

Comment:
I think you should use the same signatures as Apollo hooks have.

`[mutationFunction, { loading, error }]`

I would not make `setError` public. I think the error this hook has should represent the possible errors in the mutations.

I would consider whether the returned error should not be derived from the errors in the two mutation hooks directly without storing them within another state. This is a more minor point imo.